ecosystem/typescript/sdk_v2/src/type_utils/abi.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * The ABI JSON related types.
+ */
+
+export interface ABIRoot {
```

Comment:
We can't, because that `MoveModule` type declare property as a dynamic array, like: `exposed_functions: Array<MoveFunction>`, so it will be just a `Array<MoveFunction>`, instead of a full JSON available for type system, because TypeScript would think this `Array` type is mutable.